### PR TITLE
Adds `freeze` option for opting-out of Object.freeze usage

### PR DESCRIPTION
**Pull request by bigtimebuddy**
Addresses #1693

### Added

This PR adds the config option `freeze` (defaults to `true`). When set to `false`, it will no longer use `Object.freeze` for imported namespaces (`import * as foo from ‘foo-module’`). 

Provides a convenient opt-out for namespace protection in cases where it is undesirable for projects. In the attached issue, the use-case is injecting deprecation warnings for removed namespace members.

### CODE REVIEW

File: bin/src/index.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ const command = minimist( process.argv.slice( 2 ), {
 		h: 'help',
 		i: 'input',
 		l: 'legacy',
+		z: 'frozen',
```
Comment (TrySound) [inline bin/src/index.js]: I don't think quite specific feature should have an alias. Writing the full version will be less confusing.

Comment (bigtimebuddy) [reply bin/src/index.js]: Makes sense 👍

File: bin/src/run/mergeOptions.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ export default function mergeOptions ( config, command ) {
 	const inputOptions = {
 		input: getOption('input'),
 		legacy: getOption('legacy'),
+		frozen: getOption('frozen'),
```
Comment (lukastaegert) [inline bin/src/run/mergeOptions.js]: As freezing is something that only influences the form of the output, I would consider this to be an output option only and not something that belongs to the general options as well, what do you think? Having this duplicity between input and output options seems to me rather confusing.

Comment (bigtimebuddy) [reply bin/src/run/mergeOptions.js]: I agree, good suggestion.

**Comment by bigtimebuddy** [comment]
Definitely will add unit-test(s). Thanks for reviewing @lukastaegert.

**Comment by bigtimebuddy** [comment]
I added a unit-test for `freeze` (also, renamed from `frozen` for a more intuitive name without the tense-change).

**Comment by lukastaegert** [comment]
Thanks, great test! There was just one issue left with the handling of options, see my comment. Otherwise I think this is fine now and can go into the next version.

**Comment by bigtimebuddy** [comment]
Nice work! Thanks for this

**Comment by lukastaegert** [comment]
I have started to assemble a new branch for the next release but it will take another few days until I release it because I there are a few more PRs I want to have in there which are not ready yet.

**Pull request merged**